src/ScottPlot5/ScottPlot/Control/Interactions/StandardInteractions.cs

```diff
@@ -18,27 +25,51 @@ public StandardInteractions(IPlotControl control)
         Control = control;
     }
 
-    public virtual void MouseDown(Pixel pixel, MouseButton button)
+    public virtual void MouseDown(Pixel pixel, MouseButton button, IEnumerable<Key> keys)
+    {
+        if (button == InputMap.ZoomIn)
+        {
+            ZoomIn(pixel, keys);
+        }
+        else if (button == InputMap.ZoomOut)
+        {
+            ZoomOut(pixel, keys);
+        }
+        else
+        {
+            return;
+        }
+    }
+
+    private void ZoomIn(Pixel pixel, IEnumerable<Key> keys)
     {
+        double xFrac = InputMap.IsLockedX(keys) ? 1 : ZoomInFraction;
+        double yFrac = InputMap.IsLockedY(keys) ? 1 : ZoomInFraction;
 
+        Plot.MouseZoom(xFrac, yFrac, pixel);
+        Control.Refresh();
+    }
+
+    private void ZoomOut(Pixel pixel, IEnumerable<Key> keys)
+    {
+        double xFrac = InputMap.IsLockedX(keys) ? 1 : ZoomOutFraction;
+        double yFrac = InputMap.IsLockedY(keys) ? 1 : ZoomOutFraction;
+
+        Plot.MouseZoom(xFrac, yFrac, pixel);
+        Control.Refresh();
     }
 
     public virtual void MouseUp(Pixel pixel, MouseButton button, bool endDrag)
     {
-        switch (button)
+        if (button == InputMap.ClickDragZoomRectangle)
```

Comment:
Can this be `InputMap.IsZoomingRectangle`?

Reply:
It makes a little more sense when you see it in context

```cs
public class InputMap
{
    public MouseButton? ClickDragPan = null;
    public MouseButton? ClickDragZoom = null;
    public MouseButton? ClickDragZoomRectangle = null;
    public MouseButton? DoubleClick = null;
    public MouseButton? ZoomIn = null;
    public MouseButton? ZoomOut = null;

    public Key? LockHorizontalAxis = null;
    public Key? LockVerticalAxis = null;
    public Key? PanZoomRectangle = null;
```

It would be more verbose, but maybe this could be better communicated by renaming to

```cs
public MouseButton? ClickDragPanButton = null;
public MouseButton? ClickDragZoomButton = null;
public MouseButton? ClickDragZoomRectangleButton = null;
public MouseButton? DoubleClickButton = null;
public MouseButton? ZoomInButton = null;
public MouseButton? ZoomOutButton = null;

public Key? LockHorizontalAxisKey = null;
public Key? LockVerticalAxisKey = null;
public Key? PanZoomRectangleKey = null;
```